cmd/gateway-api/main.go

```diff
@@ -17,9 +46,123 @@ func main() {
+	// Check if we are running inside kubernetes. Hence try authenticating with service token
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		utils.AviLog.Warnf("We are not running inside kubernetes cluster. %s", err.Error())
+	} else {
+		utils.AviLog.Info("We are running inside kubernetes cluster. Won't use kubeconfig files.")
+		kubeCluster = true
+	}
+	if !kubeCluster {
+		cfg, err = clientcmd.BuildConfigFromFlags(masterURL, kubeconfig)
+		utils.AviLog.Infof("master: %s", masterURL)
+		if err != nil {
+			utils.AviLog.Fatalf("Error building kubeconfig: %s", err.Error())
+		}
+	}
+
+	akoControlConfig := gwlib.AKOControlConfig()
+	var gwApiClient *gwApi.Clientset
+
+	gwApiClient, err = gwApi.NewForConfig(cfg)
+	if err != nil {
+		utils.AviLog.Fatalf("Error building gateway-api clientset: %s", err.Error())
+	}
+	akoControlConfig.SetGatewayAPIClientset(gwApiClient)
+
+	//dynamicClient, err := lib.NewDynamicClientSet(cfg)
+	//if err != nil {
+	//	utils.AviLog.Warnf("Error while creating dynamic client %v", err)
+	//}
+
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		utils.AviLog.Fatalf("Error building kubernetes clientset: %s", err.Error())
+	}
+
+	utils.AviLog.Infof("Successfully created kube client for ako-gateway-v2")
+
+	akoControlConfig.SetEventRecorder(lib.AKOGatewayEventComponent, kubeClient, false)
+	pod, err := kubeClient.CoreV1().Pods(utils.GetAKONamespace()).Get(context.TODO(), os.Getenv("POD_NAME"), metav1.GetOptions{})
+	if err != nil {
+		utils.AviLog.Warnf("Error getting AKO pod details, %s.", err.Error())
+	}
+	akoControlConfig.SaveAKOPodObjectMeta(pod)
+
+	registeredInformers, err := gwlib.InformersToRegister(kubeClient)
+	if err != nil {
+		utils.AviLog.Fatalf("Failed to initialize informers: %v, shutting down AKO-Infra, going to reboot", err)
+	}
+
+	informersArg := make(map[string]interface{})
+
+	utils.NewInformers(utils.KubeClientIntf{ClientSet: kubeClient}, registeredInformers, informersArg)
+	//lib.NewDynamicInformers(dynamicClient, false)
+
+	informers := k8s.K8sinformers{Cs: kubeClient}
+	c := gwk8s.SharedGatewayController()
+	stopCh := utils.SetupSignalHandler()
+	ctrlCh := make(chan struct{})
+	quickSyncCh := make(chan struct{})
+
+	err = k8s.PopulateControllerProperties(kubeClient)
+	if err != nil {
+		utils.AviLog.Warnf("Error while fetching secret for AKO bootstrap %s", err)
+		lib.ShutdownApi()
+	}
+
+	aviRestClientPool := avicache.SharedAVIClients()
+	if aviRestClientPool == nil {
+		utils.AviLog.Fatalf("Avi client not initialized")
+	}
+
+	if aviRestClientPool != nil && !avicache.IsAviClusterActive(aviRestClientPool.AviClient[0]) {
+		akoControlConfig.PodEventf(corev1.EventTypeWarning, lib.AKOShutdown, "Avi Controller Cluster state is not Active")
+		utils.AviLog.Fatalf("Avi Controller Cluster state is not Active, shutting down AKO")
+	}
+
+	akoControlConfig.SetLicenseType(aviRestClientPool.AviClient[0])
+
+	//config map
+
+	//node cache
```

Comment:
node cache is not required since we are not dealing with `vrf`.

Reply:
removed comments